R/junction-read-count.R

```diff
@@ -99,18 +135,33 @@ calculatePromoterReadCounts <- function(promoterAnnotationData, junctionFilePath
   if (any(!checkFile)) {
     stop(paste0('Error: Please specify valid junction file paths. The following file does not exist: ', junctionFilePaths[!checkFile]))
   }
+  
+  if (is.null(junctionFileLabels)) {
+    junctionFileLabels <- paste0('s', 1:length(junctionFilePaths))
+  }
 
   if (numberOfCores == 1) {
     promoterReadCounts <- lapply(junctionFilePaths, calculateJunctionReadCounts, exonRanges = reducedExonRanges(promoterAnnotationData),
-                                 intronRanges = annotatedIntronRanges(promoterAnnotationData), junctionType = junctionType)
-  } else {
+                                 intronRanges = annotatedIntronRanges(promoterAnnotationData), junctionType = junctionType, genome = genome)
+  } else if (numberOfCores > 1 & .Platform$OS.type != 'windows') {
```

Comment:
you can try the same as above to avoid having the same code for windows and other systems